src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -75,7 +75,11 @@ default boolean hasCraftingRemainingItem() {
      *         decide.
      */
     default int getBurnTime(@Nullable RecipeType<?> recipeType) {
-        return self().getItem().getBurnTime(self(), recipeType);
+        int ret = self().getItem().getBurnTime(self(), recipeType);
+        if (self().isEmpty() && ret >= 0) {
+            return 0;
+        }
```

Comment:
I'm pretty sure the `isEmpty` check can come first. Also, there shouldn't be an additional constraint on the return value since that says an empty item with no burn time will still trigger the event, even though the event probably already handles it.

You should also handle the docs since `-1` is now not the value for default behavior, it's just not overriding the method.